app/models/invite_redeemer.rb

```diff
@@ -36,6 +36,18 @@ def self.create_user_from_invite(invite, username, name, password=nil)
       user.password = password
     end
 
+    if SiteSetting.must_approve_users?
+      if invite.invited_by.staff?
+        user.approved = true
+        user.approved_by_id = invite.invited_by_id
+        user.approved_at = Time.now
+      end
+    else
+      user.approved = true
+      user.approved_by_id = invite.invited_by_id
+      user.approved_at = Time.now
```

Comment:
We're duplicating code here which is not ideal. 

I think we can just switch the condition to 

`if !SiteSetting.must_aprrove_users? || (SiteSetting.must_approve_users? && invite.invited_by.staff?)`